readthedocs/v3/views.py

```diff
@@ -0,0 +1,297 @@
+    TokenAuthentication,
+)
+from rest_framework.decorators import action
+from rest_framework.metadata import SimpleMetadata
+from rest_framework.mixins import (
+    CreateModelMixin,
+    ListModelMixin,
+    RetrieveModelMixin,
+    UpdateModelMixin,
+)
+from rest_framework.permissions import IsAdminUser
+from rest_framework.renderers import BrowsableAPIRenderer
+from rest_framework.response import Response
+from rest_framework.throttling import AnonRateThrottle, UserRateThrottle
+from rest_framework.viewsets import GenericViewSet
+from rest_framework_extensions.mixins import NestedViewSetMixin
+
+from readthedocs.builds.models import Build, Version
+from readthedocs.core.utils import trigger_build
+from readthedocs.projects.models import Project
+
+from .filters import BuildFilter, ProjectFilter, VersionFilter
+from .renderer import AlphaneticalSortedJSONRenderer
+from .serializers import (
+    BuildSerializer,
+    BuildTriggerSerializer,
+    ProjectSerializer,
+    UserSerializer,
+    VersionSerializer,
+    VersionUpdateSerializer,
+)
+
+
+class APIv3Settings:
+
+    authentication_classes = (SessionAuthentication, TokenAuthentication)
+    permission_classes = (IsAdminUser,)
+    renderer_classes = (AlphaneticalSortedJSONRenderer, BrowsableAPIRenderer)
+    throttle_classes = (UserRateThrottle, AnonRateThrottle)
+    filter_backends = (filters.DjangoFilterBackend,)
+    metadata_class = SimpleMetadata
+
+
+class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, FlexFieldsMixin,
+                      ListModelMixin, RetrieveModelMixin, GenericViewSet):
+
+    # Markdown docstring is automatically rendered by BrowsableAPIRenderer.
+
+    """
+    Endpoints related to ``Project`` objects.
+
+    * Listing objects.
+    * Detailed object.
+
+    Retrieving only needed data using ``?fields=`` URL attribute is allowed.
+
+    ### Filters
+
+    Allowed via URL attributes:
+
+    * slug
+    * slug__contains
+    * name
+    * name__contains
+
+    ### Expandable fields
+
+    There are some fields that are not returned by default because they are
+    expensive to calculate. Although, they are available for those cases where
+    they are needed.
+
+    Allowed via ``?expand=`` URL attribue:
+
+    * users
+    * active_versions
+    * active_versions.last_build
+    * active_versions.last_build.confg
+
+
+    ### Examples:
+
+    * List my projects: ``/api/v3/projects/``
+    * Filter list: ``/api/v3/projects/?name__contains=test``
+    * Retrieve only needed data: ``/api/v3/projects/?fields=slug,created``
+    * Retrieve specific project: ``/api/v3/projects/{project_slug}/``
+    * Expand required fields: ``/api/v3/projects/{project_slug}/?expand=active_versions``
+    * Translations of a project: ``/api/v3/projects/{project_slug}/translations/``
+    * Subprojects of a project: ``/api/v3/projects/{project_slug}/subprojects/``
+    * Superproject of a project: ``/api/v3/projects/{project_slug}/superproject/``
+    """
```

Comment:
Would be good to show how to get more results here. Is it `?count=25`? I couldn't figure it out by guessing.

Reply:
We use `limit=` and `offset=`, as in APIv2.